airbyte-scheduler/models/src/main/java/io/airbyte/scheduler/models/AttemptWithJob.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.scheduler.models;
+
+public class AttemptWithJob {
+
+  private final Attempt attempt;
+  private final Job job;
+
+  public Job getJob() {
```

Comment:
I feel like using the type of `Job` here is maybe a bit dangerous. Since the `Job` object has a list of `Attempt`s that I do not think will get populated properly in how you are querying using. We use special `ORDER BY` in the existing list job queries to make sure that `Attempt`s are added to the `Job` struct in the correct order. The `ORDER BY` you are introducing, I think, is going to break that guarantee.

Could we just deconstruct the information that we need from the Job here instead of trying to reuse this object? I know it feels like we could reuse `Job` here but I think it'll be confusing.

Reply:
That or if we do reuse Job, I guess we could explicitly set the attempts field to null, but I think that might be confusing and put a big javadoc comment on the method explaining why.

Reply:
I see what you mean. I agree that it could be confusing for someone that tries use this method or changes some logic that uses the result of this method, and they expect the `Job` object to have a list of attempts, since all other instances of `Job` do. 

I think deconstructing the info that we need sounds like a good approach, since it uses java typing to enforce that people won't try to access the `attempts` field of this result, rather than relying on people to read javadocs 👍